src/engine/GroupBy.cpp

```diff
@@ -538,38 +633,29 @@ std::optional<IdTable> GroupBy::computeGroupByForFullIndexScan() {
         "not supported."};
   }
 
-  // Prepare the `result`
-  size_t numCols = numCounts + 1;
   _subtree->getRootOperation()->updateRuntimeInformationWhenOptimizedOut({});
 
-  // A nested lambda that computes the actual result. The outer lambda is
-  // templated on the number of columns (1 or 2) and will be passed to
-  // `callFixedSize`.
-  auto doComputationForNumberOfColumns = [&]<int NUM_COLS>() mutable {
-    auto ignoredRanges =
-        getIndex().getImpl().getIgnoredIdRanges(permutationEnum.value()).first;
-    const auto& permutation =
-        getExecutionContext()->getIndex().getPimpl().getPermutation(
-            permutationEnum.value());
-    auto table = permutation.getDistinctCol0IdsAndCounts(cancellationHandle_);
-    if (NUM_COLS == 1) {
-      table.setColumnSubset({{0}});
-    }
-    // TODO<joka921> This is only semi-efficient.
-    auto end = std::ranges::remove_if(table, [&ignoredRanges](const auto& row) {
-      return std::ranges::any_of(ignoredRanges,
-                                 [id = row[0]](const auto& pair) {
-                                   return id >= pair.first && id < pair.second;
-                                 });
+  auto ignoredRanges =
+      getIndex().getImpl().getIgnoredIdRanges(permutationEnum.value()).first;
+  const auto& permutation =
+      getExecutionContext()->getIndex().getPimpl().getPermutation(
+          permutationEnum.value());
+  auto table = permutation.getDistinctCol0IdsAndCounts(cancellationHandle_);
+  if (numCounts == 0) {
+    table.setColumnSubset({{0}});
+  }
+  // TODO<joka921> This is only semi-efficient.
+  auto end = std::ranges::remove_if(table, [&ignoredRanges](const auto& row) {
+    return std::ranges::any_of(ignoredRanges, [id = row[0]](const auto& pair) {
+      return id >= pair.first && id < pair.second;
     });
-    table.resize(end.begin() - table.begin());
-    return table;
-  };
+  });
+  table.resize(end.begin() - table.begin());
 
   // TODO<joka921> This optimization should probably also apply if
   // the query is `SELECT DISTINCT ?s WHERE {?s ?p ?o} ` without a
   // GROUP BY, but that needs to be implemented in the `DISTINCT` operation.
-  return ad_utility::callFixedSize(numCols, doComputationForNumberOfColumns);
```

Comment:
This is a minor unrelated change that reduces complexity and reduces compilation times slightly

Reply:
Alright, maybe you can extract a smaller preparation PR that factors out all the changes that are not directly lazy
(factoring oyut common code, this thing, changes in the HashMap Optimization, etc.